### PR TITLE
feat: Ensure teams with conflicting slugs owned by the user are migrated(handled in backend, frontend already had this restriction)

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

This PR adds backend validation to prevent slug conflicts when creating organizations. It ensures that if a user owns a team with the same slug as the organization they're creating, that team is automatically marked for migration.

**Key Changes:**
1. Added `findOwnedTeamsByUserId` method to `TeamRepository` to query teams where user is OWNER or ADMIN
2. Created `buildTeamsAndInvites` method in `BaseOnboardingService` that:
   - Automatically detects teams with same slug as organization
   - Marks conflicting teams for migration (`isBeingMigrated: true`)
   - Filters empty team names and invite emails
3. Refactored `listOwnedTeamsHandler` to use the new repository method instead of direct Prisma queries
4. Added comprehensive tests for slug conflict scenarios

This ensures backend validation even if frontend validation is bypassed, preventing slug conflicts during organization creation. All classes inheriting from `BaseOnboardingService` automatically get this validation.

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

**Test Scenario 1: Automatic Migration Detection**
1. Create a team with slug "acme-corp" as an OWNER
2. Attempt to create an organization with slug "acme-corp"
3. Expected: The team should be automatically added to the migration list with `isBeingMigrated: true`

**Test Scenario 2: Existing Team in List**
1. Create a team with slug "acme-corp" as an OWNER
2. Include that team in the onboarding request with `isBeingMigrated: false`
3. Expected: The team should be updated to have `isBeingMigrated: true`

**Test Scenario 3: Non-Conflicting Slugs**
1. Create a team with slug "different-team" as an OWNER
2. Create an organization with slug "acme-corp"
3. Expected: The team should NOT be added to the migration list

**Test Scenario 4: Member Role**
1. Create a team with slug "acme-corp" where user is only a MEMBER (not OWNER/ADMIN)
2. Attempt to create an organization with slug "acme-corp"
3. Expected: The team should NOT be migrated (user doesn't own it)

**Environment Variables:**
- Standard Cal.com environment variables
- No additional configuration needed

## Human Review Checklist

Please pay special attention to:

1. **Query Logic Equivalence**: Verify that `TeamRepository.findOwnedTeamsByUserId` returns the same results as the original Prisma query in `listOwnedTeamsHandler` (filters organizations, checks OWNER/ADMIN roles, requires accepted membership)

2. **Automatic Migration Behavior**: Confirm that automatically adding teams to the migration list is the desired behavior and won't cause unexpected side effects in production

3. **Method Signature Change**: The `filterTeamsAndInvites` method was changed to async `buildTeamsAndInvites`. Verify all callers have been updated (both `BillingEnabledOrgOnboardingService` and `SelfHostedOnboardingService`)

4. **Test Coverage**: Review test scenarios, especially edge cases around:
   - Teams already marked for migration
   - Non-conflicting slugs
   - User role permissions (OWNER/ADMIN vs MEMBER)

5. **Repository Pattern**: Verify that instantiating `TeamRepository` directly in `BaseOnboardingService` follows the codebase's dependency injection patterns

---

**Link to Devin run:** https://app.devin.ai/sessions/b90a924a3ffb4a0cb03cb0c79f5f98e0  
**Requested by:** hariom@cal.com (@hariombalhara)

### CODE REVIEW

File: packages/features/ee/organizations/lib/service/onboarding/BaseOnboardingService.ts [hunk 1]
```diff
@@ -143,8 +144,53 @@ export abstract class BaseOnboardingService implements IOrganizationOnboardingSe
...
+      {
+        id: conflictingTeam.id,
+        name: conflictingTeam.name,
+        isBeingMigrated: true,
+        slug: conflictingTeam.slug,
+      },
+    ];
+  }
+
+  protected async buildTeamsAndInvites(
```
Comment (hariombalhara) [inline packages/features/ee/organizations/lib/service/onboarding/BaseOnboardingService.ts:185]: Renamed _filterTeamsAndInvites_ -> _buildTeamsAndInvites_

File: packages/features/ee/organizations/lib/service/onboarding/BaseOnboardingService.ts [hunk 2]
```diff
@@ -143,8 +144,53 @@ export abstract class BaseOnboardingService implements IOrganizationOnboardingSe
...
+      if (!existingTeam.isBeingMigrated) {
+        const updatedTeams = [...teams];
+        updatedTeams[existingTeamIndex] = {
+          ...existingTeam,
+          isBeingMigrated: true,
+        };
+        return updatedTeams;
+      }
+      return teams;
+    }
```
Comment (Udit-takkar) [inline packages/features/ee/organizations/lib/service/onboarding/BaseOnboardingService.ts]: imo 

```
 const conflictingTeam = ownedTeams.find((team) => team.slug === orgSlug);

  if (!conflictingTeam) {
    return teams;
  }

  const existingTeam = teams.find((t) => t.id === conflictingTeam.id);

  if (existingTeam) {
    // Already marked for migration - no change needed
    if (existingTeam.isBeingMigrated) {
      return teams;
    }
    
    // Mark existing team for migration
    return teams.map((team) =>
      team.id === conflictingTeam.id 
        ? { ...team, isBeingMigrated: true }
        : team
    );
  }

```
this is more easily readable

Comment (hariombalhara) [reply packages/features/ee/organizations/lib/service/onboarding/BaseOnboardingService.ts]: 100pc agree !!

**Review by Udit-takkar — APPROVED** [review]
Looks good now
